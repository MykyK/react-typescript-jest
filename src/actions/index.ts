import { EDIT_USER } from "../constants";
import { IEditUserAction, IUser } from "../interfaces";

export const actionEditUser = (user: IUser): IEditUserAction => ({
  type: EDIT_USER,
  payload: {
    user
  }
});
