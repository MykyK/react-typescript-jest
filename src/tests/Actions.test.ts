import configureStore from 'redux-mock-store';
import { actionEditUser } from '../actions';



const mockStore = configureStore();
const store = mockStore();
const user = {
  "id": 1,
  "name": "Leanne Graham",
  "username": "Bret",
  "email": "Sincere@april.biz",
  "is_admin": true,
  "age": 21,
  "address": {
    "street": "Kulas Light",
    "suite": "Apt. 556",
    "city": "Gwenborough",
    "zipcode": "92998-3874"
  },
}

describe('select_actions', () => {
  beforeEach(() => { // Runs before each test in the suite
    store.clearActions();
  });

  describe('actionEditUser', () => {
    test('Dispatches the correct action and payload', () => {
      const expectedActions =
    [
      {
        'payload': {user},
        'type': 'EDIT_USER',
      },
    ]

      store.dispatch(actionEditUser(user));
      expect(store.getActions()).toEqual(expectedActions);
    });
  });
});
