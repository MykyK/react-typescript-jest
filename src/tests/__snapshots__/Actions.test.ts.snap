// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_actions actionEditUser Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": Object {
      "user": Object {
        "address": Object {
          "city": "Gwenborough",
          "street": "Kulas Light",
          "suite": "Apt. 556",
          "zipcode": "92998-3874",
        },
        "age": 21,
        "email": "Sincere@april.biz",
        "id": 1,
        "is_admin": true,
        "name": "Leanne Graham",
        "username": "Bret",
      },
    },
    "type": "EDIT_USER",
  },
]
`;
